app/streams/publishing_api_consumer.rb

```diff
@@ -11,24 +11,56 @@ def process(message)
 private
 
   def process_message(message)
-    content_id = message.payload['content_id']
-    base_path = message.payload['base_path']
+    content_id = message.payload.fetch('content_id')
+    base_path = message.payload.fetch('base_path')
+    payload_version = message.payload.fetch('payload_version')
     locale = message.payload['locale'] || 'en'
 
     item = Dimensions::Item.by_natural_key(content_id: content_id, locale: locale)
+
     if item
-      handle_existing(item, base_path, message.delivery_info.routing_key)
+      existing_payload_version = item.publishing_api_payload_version || 0
+
+      if payload_version > existing_payload_version
+        handle_existing(
+          item: item,
+          base_path: base_path,
+          routing_key: message.delivery_info.routing_key,
+          payload_version: payload_version
+        )
+      else
+        # Skip out of date and repeated messages from the publishing API.
```

Comment:
What about wrapping the comment in a string, and append it as additional information to the event? I think it is quite descriptive and would help when looking the logs. I am not a fan of big comments in code.

Reply:
To the log event you mean?

Reply:
do you mean something like this?

```ruby
        logger.info %Q{Skipping message from publishing API: published with #{payload_version} but we've already received a message with #{item.publishing_api_payload_version}.
This is to ensure that our "latest" record is actually the latest.
RabbitMQ guarantees at-least-once delivery when using acknowledgements, so messages can be duplicated, plus we can't guarantee the order of processing because we can have multiple consumers working in parallel.}
```

I'd prefer the original or not including the comment at all, I think this clutters up the logs for not much benefit. The reason I added a the comment is because it's non-obvious behaviour that may not make sense when looking at the code

Reply:
I was thinking in crafting a message using `heredocs`, and probably extracting a method with `what` we are actually doing; let's have a second iteration, as this is a big PR, ad a good step forward.